dockerfiles/image.sh

```diff
@@ -69,6 +69,25 @@ finalize_platform_core_image() {
     docker commit $CONTAINER hpccsystems/platform-core:$GIT_BRANCH-$MODE-$crc
     docker stop $CONTAINER
     docker rm $CONTAINER
+
+    if [ "$MODE" = "debug" ]; then
+        # Add sources
+        echo "--- Adding sources to '$image_name' image ---"
+        local CONTAINER=$(docker run -d \
+            --mount source=$ROOT_DIR,target=/hpcc-dev/HPCC-Platform-local,type=bind,readonly \
+            --mount source=$HPCC_BUILD,target=/hpcc-dev/build,type=volume \
+            hpccsystems/platform-core:$GIT_BRANCH-$MODE-$crc "tail -f /dev/null")
+        docker exec --user root --workdir /hpcc-dev/HPCC-Platform $CONTAINER /bin/bash -c "chown -R hpcc:hpcc /hpcc-dev/HPCC-Platform"
+        docker exec --workdir /hpcc-dev/HPCC-Platform $CONTAINER /bin/bash -c "git config --global --add safe.directory /hpcc-dev/HPCC-Platform-local"
+        docker exec --workdir /hpcc-dev/HPCC-Platform $CONTAINER /bin/bash -c "cp -r ../HPCC-Platform-local/.git ./.git"
```

Comment:
minor: I know it's temporary, but .git (with history etc.) can be quite big/time consuming to copy.
I think you could instead copy everything with git archive like this:
```
cd ../HPCC-Platform-local
git archive --format=tar HEAD | tar xf - -C /hpcc-dev/HPCC-Platform
git submodule foreach --recursive 'git archive --format=tar HEAD | (cd /hpcc-dev/HPCC-Platform; mkdir -p $path; cd $path; tar xf -)'
```